File: e2e/playwright-typescript/tests/create-task/create-task.uat.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.TASK_MANAGEMENT);
  await allure.feature('Create task');
});

test.describe('Create task', () => {
  test('should create task', { tag: '@uat' }, async ({ step, validate }) => {
    // given
    const context = new TaskContext();

    // when
    await step.navigation.openMainPage();
    await step.tasks.openCreateTaskForm();
    await step.tasks.createTask.fillForm(context.createTaskData());
    await step.tasks.createTask.submitForm();

    // then
    await validate.tasks.hasTask(context.title);

    // when
    await step.tasks.openTaskInfoForm(context.title);

    // then
    await validate.task.data(context.createTaskData());
  });
});

File: e2e/capybara-ruby/spec/create_task/create_task_uat_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'Create task', type: :system, uat: true,
                             epic: AllureEpic::TASK_MANAGEMENT,
                             feature: 'Create task',
                             tms: '100' do
  it 'should create task' do
    # given
    context = TaskContext.new

    # when
    step.navigation.open_main_page
    step.tasks.open_create_task_form
         .fill_form(context.create_task_data)
         .submit_form

    # then
    validate.tasks.has_task(context.title)

    # when
    step.tasks.open_task_info_form(context.title)

    # then
    validate.task.data(context.create_task_data)
  end
end
